explorer/deploy/webAppCreator.ts

```diff
@@ -543,4 +565,25 @@ class WebsiteStep extends WebAppCreatorStepBase {
         }
     }
 
+    //Implements new Service principal model for ACR container registries while maintaining old admin enabled use
+    private async acquireRegistryLoginCredentials(): Promise<void> {
+        if (this._serverPassword && this._serverUserName) { return; }
+
+        if (this._registry.adminUserEnabled) {
+            const client = new ContainerRegistryManagementClient(this.azureAccount.getCredentialByTenantId(this._imageSubscription.tenantId), this._imageSubscription.subscriptionId);
+            const resourceGroup: string = this._registry.id.slice(this._registry.id.search('resourceGroups/') + 'resourceGroups/'.length, this._registry.id.search('/providers/'));
+            try {
+                let creds = await client.registries.listCredentials(resourceGroup, this._registry.name);
+                this._serverPassword = creds.passwords[0].value;
+                this._serverUserName = creds.username;
+                return;
+            } catch (error) {
+                vscode.window.showErrorMessage(error.message);
+                throw new Error(('Azure Container Registry Admin is not enabled'));
+            }
+        }
+        vscode.window.showErrorMessage('Azure App service currently only supports running images from Azure Container Registries with admin enabled');
```

Comment:
Again, you can't make that assumption.  Does the error that you get back in this scenario have a particular code that you can check for in order to throw a more specific error in this case?

Reply:
I could check for whether the list failed because of admin was not enabled specifically and then throw accordingly. As for the Azure App service does not support... etc I have to make the assumption as the only way to identify this error is to go into the azure portal and note that your web app fails due to admin not being enabled, no error is actually given to the user from the API as long as credentials are valid.

Reply:
I'm confused - you're throwing this error specifically when you get an exception.  Can you not check for the error code?